### ISSUE TITLE
Testcafe 2.1.0 support?

### ISSUE THREAD

**Issue by MPriess** [Arthy000/gherkin-testcafe#108]
Does the latest release of gherkin-testcafe 5.5.1 support testcafe 2.1.0? 

I had a look at the [package.json](https://github.com/Arthy000/gherkin-testcafe/blob/v5.5.1/package.json) but it seems like testcafe version 1.20 is referenced.

But the documentation mention:

```
Upgrading from version 1.x
With TestCafé version 2.0, this package has introduced some breaking changes to it's API. These changes help this package be more future-proof in terms of upcoming features.

When upgrading from version 1 to version 2, keep in mind that the following things have changed...
```

Now I'm puzzled.

**Comment by MPriess** [comment]
Adding the testcafe version 2.1.0 to the package.json lead to an error:


```
"devDependencies": {
    "gherkin-testcafe": "5.5.1",
    "testcafe": "2.1.0"
  }
 }
```


➜  testcases git:(renovate/testcafe-2.x) ✗ npm install             
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: testing@1.0.0
npm ERR! Found: testcafe@2.1.0
npm ERR! node_modules/testcafe
npm ERR!   dev testcafe@"2.1.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer testcafe@"~1.20.0" from gherkin-testcafe@5.5.1
npm ERR! node_modules/gherkin-testcafe
npm ERR!   dev gherkin-testcafe@"5.5.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/michael.priess/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/michael.priess/.npm/_logs/2022-12-07T14_09_53_323Z-debug-0.log

So I decided to overwrite the testcafe depependency provided by gherkin-testcafe:

```
 "devDependencies": {
    "gherkin-testcafe": "5.5.1",
    "testcafe": "2.1.0"
  },
  "overrides": {
    "gherkin-testcafe": {
      "testcafe": "$testcafe"
    }
  }
```

**Comment by Arthy000** [comment]
Hi @MPriess 

Indeed this is a mistake. I think the original author of the project meant TestCafe@0.2.0
Up until recently, TestCafe was still in version 1.X so this wasn't too big a deal, but now I have to fix this.

Regarding support for TestCafe V2, it is coming. At the time I tested version 2.0 and it worked as is (meaning, no update needed), but they usually roll out 3-4 versions in a matter of weeks, so I tend to wait a bit before working on the update.

TLDR: I'll fix the doc and I'll make GTC work with TC 2.x

**Comment by Arthy000** [comment]
In defense of the original author, TestCafe versioning has always been a bit weird.

**Comment by Arthy000** [comment]
Hi @MPriess, 

As I expected there were no required changes to make GTC work with TC@2.1.0.
To make it clearer, I started maintaining a compatibility matrix. I also took the time to properly setup the peer dependency in my package.json: you can now use any version between 1.20.0 and 2.1.0 (included).

I did however set it up so that when a new version is released I have to update the peer dependencies (and the compatiblity matrix), which means users of GTC _can_ install a newer version of TC (you can ignore unmatched peer dependencies, in `yarn` it's only a warning, in `npm` you have to use a specific option, at least according to the output you pasted)

Anyway, you can now install `gherkin-testcafe@5.5.2` and use it with `testcafe@2.1.0` without errors or warnings.

**Comment by MPriess** [comment]
Integrated 5.5.2 today. Works as expected. Great work 👍
